src/Chainweb/Pact/PactService.hs

```diff
@@ -128,46 +128,48 @@ initPactService'
     -> PactServiceConfig
     -> PactServiceM cas a
     -> IO (T2 a PactServiceState)
-initPactService' ver cid chainwebLogger bhDb pdb sqlenv config act = do
-    checkpointEnv <- initRelationalCheckpointer initialBlockState sqlenv cplogger ver cid
-    let !rs = readRewards
-        !initialParentHeader = ParentHeader $ genesisBlockHeader ver cid
-        !pse = PactServiceEnv
-                { _psMempoolAccess = Nothing
-                , _psCheckpointEnv = checkpointEnv
-                , _psPdb = pdb
-                , _psBlockHeaderDb = bhDb
-                , _psGasModel = getGasModel
-                , _psMinerRewards = rs
-                , _psReorgLimit = fromIntegral $ _pactReorgLimit config
-                , _psOnFatalError = defaultOnFatalError (logFunctionText chainwebLogger)
-                , _psVersion = ver
-                , _psValidateHashesOnReplay = _pactRevalidate config
-                , _psAllowReadsInLocal = _pactAllowReadsInLocal config
-                , _psIsBatch = False
-                , _psCheckpointerDepth = 0
-                , _psLogger = pactLogger
-                , _psGasLogger = gasLogger <$ guard (_pactLogGas config)
-                , _psLoggers = loggers
-                , _psBlockGasLimit = _pactBlockGasLimit config
-                }
-        !pst = PactServiceState Nothing mempty initialParentHeader P.noSPVSupport
-    runPactServiceM pst pse $ do
-
-        -- If the latest header that is stored in the checkpointer was on an
-        -- orphaned fork, there is no way to recover it in the call of
-        -- 'initalPayloadState.readContracts'. We therefore rewind to the latest
-        -- avaliable header in the block header database.
-        --
-        exitOnRewindLimitExceeded $ initializeLatestBlock (_pactUnlimitedInitialRewind config)
-        act
+initPactService' ver cid chainwebLogger bhDb pdb sqlenv config act =
```

Comment:
It's a bit disturbing that we have an `initX` function calling a `withX` function. This PR isn't responsible, but perhaps at some point we can consider renaming `initPactService'` to `withPactService`.

Reply:
I renamed `initPactService` into `runPactService`, I think, that should make it clear that it is a long-running service process.